image2tensor/src/lib.rs

```diff
@@ -53,21 +53,20 @@ pub fn calculate_buffer_size(width: u32, height: u32, precision: TensorType) ->
 }
 
 // Save the bytes into the specified TensorType format.
-fn save_bytes(arr: &[u8], tt: TensorType) -> Vec<u8> {
+fn save_bytes(buffer: &[u8], tt: TensorType) -> Vec<u8> {
     let mut out: Vec<u8> = vec![];
 
-    for i in 0..arr.len() {
+    for &byte in buffer {
         // Split out the bytes based on the TensorType.
-        let bytes: Vec<u8>;
-        bytes = match tt {
-            TensorType::F16 => (arr[i] as f32).to_ne_bytes().to_vec(),
-            TensorType::F32 => (arr[i] as f32).to_ne_bytes().to_vec(),
-            TensorType::U8 => (arr[i]).to_ne_bytes().to_vec(),
-            TensorType::I32 => (arr[i] as i32).to_ne_bytes().to_vec(),
+        let ne_bytes = match tt {
+            TensorType::F16 => (byte as f32).to_ne_bytes().to_vec(),
```

Comment:
Pretty sure this will blow up backends if it ever gets used, since they will not be expecting 4 bytes per element in the tensor if tensor type is f16. We should probably assert an error if someone requests those tensor types.

Reply:
ok, see 27f0b1c